invoiceregistry_test.go

```diff
@@ -320,6 +320,39 @@ func TestNoSubscriberFail(t *testing.T) {
 	c.checkHtlcFailed(ResultNoAcceptSubscriber)
 }
 
+// TestGracePeriodForSubscriber asserts that a new set is started even if the last subscriber has recently disconnected.
+// After the grace period, we should not accept any new sets
+func TestGracePeriodForSubscriber(t *testing.T) {
+	defer test.Timeout()()
+	t.Parallel()
+
+	// If we reach the accept timeout, it means we've accepted the set_id
+	c := newRegistryTestContext(t, optAcceptTimeout(0))
+
+	// Add invoice.
+	invoice, preimage := c.createInvoice(1, time.Hour)
+
+	// A subscriber connects and disconnects
+	_, acceptCancel := c.subscribeAccept()
+	acceptCancel()
+
+	c.notifyHtlc(preimage.Hash(), invoice.PaymentAddr)
+
+	// The set was accepted but we reach accept timeout
+	c.checkHtlcFailed(ResultAcceptTimeout)
+
+	// Grace period is finished
+	gracePeriodWithoutSubscribers = 0
```

Comment:
Is it really necessary to change this value which is mostly a constant?

Reply:
It was simply to avoid using time.sleep()

But related to https://github.com/bottlepay/lnmux/pull/95#discussion_r989923982

Using a clock will solve the problem